### DISCUSSION TITLE
Better code blocks for super notes

### DISCUSSION THREAD

**Discussion by adamhotep** [standardnotes/forum#3465] [Suggestions]
Code blocks in super notes currently have syntax highlighting that you cannot change. I tend to write code blocks for other programming languages. My most common use for code blocks is just to get fixed-width text (it's not code). The syntax highlighting is distracting. I'd like to be able to specify the language. Defaulting to "none" seems best imho, but if you stick with whatever you've got, be sure to allow users to turn it off.

I see a lot of markdown is supported in super notes, but not fence-post code blocks, which would allow specifying the syntax highlighting. For example:

````
```bash
#!/bin/sh
if [ 7 -gt 4 ]; then
  echo math
fi
```
````

On Github, this renders as:

```bash
#!/bin/sh
if [ 7 -gt 4 ]; then
  echo math
fi
```

(Also note that I was able to seamlessly write the above markdown code block in markdown code by using a quad-backtick around my markdown-as-code block. This is useful! I'm not sure how that should manifest in super notes though.)

**Comment by DarkArc** [comment]
There should be a thing that floats that lets you set the language.

**Comment by adamhotep** [comment]
@DarkArc: That would be great. Where is it? I don't see that on the web app or on Android no matter where I click/tap:

![image](https://user-images.githubusercontent.com/6454774/235945203-45cda095-285b-4c67-8771-c154c1db90d1.png)

**Comment by DarkArc** [comment]
![image](https://user-images.githubusercontent.com/778012/235945926-f3682374-cb6b-4121-8f96-d3e7563ab248.png)
It _should_ look like this, at least it does on the desktop app (maybe it's broken on the web app?).

**Comment by adamhotep** [comment]
Ah, then maybe this is a bug on the two platforms I'm using (and a feature request for markdown fenced code block support)

**Comment by adamhotep** [comment]
The bug part appears fixed now. The syntax drop-down is at the top right corner of the screen. I don't know if I had simply missed it before or if it was more recently added.

I'd still like support for opening a code block on a new line that starts with 3+ backticks and an optional syntax specification.

**Comment by alainter** [comment]
The choice of Code highlight is also quite limited and I don't see bash, php, tcl ..
At the very least some highlight support for UNIX shells, please.

**Reply by fele-systems** [reply]
I agree. They even support PowerShell, but not shell...
Most of my code in my notes are about bash, so adding a shell language option would be cool. Most of the apps that support syntax highlighting supports shell.

Also, json and yaml would be useful too.

**Comment by proton1k** [comment]
Is there a way to create a custom theme for code syntax highlighting ?
